pkg/operator/certrotation/signer.go

```diff
@@ -69,7 +75,7 @@ func (c RotatedSigningCASecret) ensureSigningCertKeyPair(ctx context.Context) (*
 
 	if needed, reason := needNewSigningCertKeyPair(signingCertKeyPairSecret.Annotations, c.Refresh, c.RefreshOnlyWhenExpired); needed {
 		c.EventRecorder.Eventf("SignerUpdateRequired", "%q in %q requires a new signing cert/key pair: %v", c.Name, c.Namespace, reason)
-		if err := setSigningCertKeyPairSecret(signingCertKeyPairSecret, c.Validity); err != nil {
+		if err := setSigningCertKeyPairSecret(signingCertKeyPairSecret, c.Validity, c.JiraComponent, c.Description); err != nil {
```

Comment:
how will this add the labels to secrets and configmaps in an existing cluster?

Reply:
atm these would be updated on rotation only. We'll need a followup to move annotation update elsewhere (possibly we should have a separate library function for that)

Reply:
> atm these would be updated on rotation only. We'll need a followup to move annotation update elsewhere (possibly we should have a separate library function for that)

We need to address this shortcoming as we merge the feature.  Upgraded clusters must be the same as freshly installed clusters.

Reply:
Fixed by introducing `NeedsTLSMetadataUpdate`